src/pages/admin/partners/components/partner-banner-form/PartnerBannerForm.tsx

```diff
@@ -0,0 +1,252 @@
+            if (
+                axios.isCancel?.(fetchError) ||
+                fetchError.name === 'CanceledError' ||
+                fetchError.name === 'AbortError'
+            ) {
+                return;
+            }
+            addToast(PARTNERS_TEXT.MESSAGE.FAIL_TO_LOAD_BANNER, ToastType.Error);
+        }
+    }, [fetchError, addToast]);
+
+    const handleTitleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = e.target.value;
+        setValues((prev) => (prev ? { ...prev, title: newValue } : null));
+        setErrors((prev) => ({
+            ...prev,
+            title: PARTNER_BANNER_VALIDATION_FUNCTIONS.validateTitle(newValue),
+        }));
+    }, []);
+
+    const handleDescriptionChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = e.target.value;
+        setValues((prev) => (prev ? { ...prev, description: newValue } : null));
+        setErrors((prev) => ({
+            ...prev,
+            description: PARTNER_BANNER_VALIDATION_FUNCTIONS.validateDescription(newValue),
+        }));
+    }, []);
+
+    const handleImageChange = useCallback((value: ImageValues | null) => {
+        setValues((prev) => (prev ? { ...prev, image: value, imageId: value ? prev.imageId : null } : null));
+    }, []);
+
+    const handleImageError = useCallback((error: string | null) => {
+        setErrors((prev) => ({
+            ...prev,
+            image: error ? error : undefined,
+        }));
+    }, []);
+
+    const handlePublish = useCallback(async () => {
+        if (!values || !isFormValid(values, errors)) return;
+
+        setIsPublishing(true);
+        try {
+            const updatedBanner = await PartnersApi.updateBanner(client, {
+                title: values.title,
+                description: values.description,
+                image: values.image,
+                imageId: values.imageId,
+            });
+
+            setValues(updatedBanner);
+            addToast(PARTNERS_TEXT.MESSAGE.BANNER_SAVED, ToastType.Success);
+        } catch (error: any) {
+            if (axios.isCancel?.(error) || error.name === 'CanceledError' || error.name === 'AbortError') {
+                return;
+            }
+            addToast(PARTNERS_TEXT.MESSAGE.FAIL_TO_UPDATE_BANNER, ToastType.Error);
+        } finally {
+            setIsPublishing(false);
+        }
+    }, [values, errors, client, addToast]);
+
+    if (isLoadingData) {
+        return (
+            <div className="partner-banner__loader">
+                <InlineLoader size={2} />
+            </div>
+        );
+    }
+
+    if (fetchError || !values) {
+        return (
+            <div className="partner-banner__error">
+                <p>{PARTNERS_TEXT.MESSAGE.FAIL_TO_LOAD_BANNER}</p>
+                <Button onClick={refetchBanner} buttonStyle="primary">
+                    {PARTNERS_TEXT.BUTTON.TRY_AGAIN}
+                </Button>
+            </div>
+        );
+    }
```

Comment:
[addition to comment ](https://github.com/ita-social-projects/VictoryCenter-Client/pull/114/files#r2534612902)
```suggestion
    if (isLoadingData) {
        return (
            <div className="partner-banner-form__loader">
                <InlineLoader size={2} />
            </div>
        );
    }

    if (fetchError || !values) {
        return (
            <div className="partner-banner-form__error">
                <p>{PARTNERS_TEXT.MESSAGE.FAIL_TO_LOAD_BANNER}</p>
                <Button onClick={refetchBanner} buttonStyle="primary">
                    {PARTNERS_TEXT.BUTTON.TRY_AGAIN}
                </Button>
            </div>
        );
    }
```